Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3569,12 +3569,19 @@ void RenderBox::computePreferredLogicalWidths(const Length& minLogicalWidth, con
         m_maxPreferredLogicalWidth = std::clamp(m_maxPreferredLogicalWidth, transferredMinLogicalWidth, transferredMaxLogicalWidth);
     }
 
-    if (maxLogicalWidth.isFixed()) {
-        auto adjustContentBoxLogicalWidth = adjustContentBoxLogicalWidthForBoxSizing(maxLogicalWidth);
-        m_maxPreferredLogicalWidth = std::min(m_maxPreferredLogicalWidth, adjustContentBoxLogicalWidth);
-        m_minPreferredLogicalWidth = std::min(m_minPreferredLogicalWidth, adjustContentBoxLogicalWidth);
-    }
+    auto usedMaxLogicalWidth = [&] {
+        // FIXME: We should be able to handle other values for the max logical width here.
+        if (maxLogicalWidth.isFixed())
+            return adjustContentBoxLogicalWidthForBoxSizing(maxLogicalWidth);
+        if (maxLogicalWidth.isMinContent())
+            return computeIntrinsicLogicalWidthUsing(maxLogicalWidth, availableLogicalWidth(), 0_lu);
```

Comment:
this looks a bit wasteful to me as in most cases we end up running preferred width computation here while we already have that value in m_minPreferredLogicalWidth -unless I am missing something.

Reply:
That's a good point. I made a couple of changes:

1.) Reshuffled the logic to compute the used min/max logical widths to the top of the function to avoid m_minPreferredLogicalWIdth getting changed by the aspect ratio logic
2.) Refactored out some of the logic used to determine when preferred widths values will get set from the value in style so we could use it here